apps/src/blockly/customBlocks/googleBlockly/spritelabBlocks.js

```diff
@@ -119,4 +131,65 @@ export const blocks = {
       flyoutToggleButton.setIcon(useDefaultIcon);
     };
   },
+
+  // Set up this block to shadow a parent block's image, if needed. This will also
+  // deserialize any parent toolbox information from the block configuration.
+  setUpBlockShadowing() {
+    // We only set up block shadowing for blocks that have a type in blockShadowingPairs.
+    if (Object.keys(blockShadowingPairs).includes(this.type)) {
+      // saveExtraState is used to serialize the parent block ID.
+      this.saveExtraState = function () {
+        return {
+          parentBlockId: this.parentBlockId,
```

Comment:
Why do we need to save this extra state? I had been thinking that we were adding the extra state to the blocks at creation, but wouldn't need to save it. This might be a good one to talk through together just so I can understand the save/reload flow we're using a little better.

Reply:
We needed to save it for the new block generation to work, i.e. when you drag out a new block from the mini toolbox. My guess is that we serialize and deserialize the existing block when that happens? So we needed to copy over the parent id.